server/src/main/java/com/testsigma/service/TestStepService.java

```diff
@@ -574,8 +577,40 @@ private List<ActionTestDataMap> getMapsList(){
 
         return actionsMap;
     }
+    public void bulkDelete(Long[] ids) throws ResourceNotFoundException {
+        for (Long id:ids){
+              this.destroy(this.find(id));
+            }
+    }
 
-
+    private Boolean checkBulkDeleteStepsHaveAllPrerequisiteSteps(List<Long> ids,List<TestStep> testStepList){
+        for (TestStep testStep:testStepList){
+            if(!ids.contains(testStep.getId())){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public List<TestStepDTO> checkForLinkedTestCases(Long[] ids) throws ResourceNotFoundException {
```

Comment:
The method name isn't appropriate. Change it to something like getPrerequisiteTestSteps